File: app/src/main/java/com/example/mvvmgithubsampleapp/di/AppComponent.java
package com.example.mvvmgithubsampleapp.di;

import com.example.mvvmgithubsampleapp.repository.ProjectsRepository;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {
        AppModule.class,
})
public interface AppComponent {

    void inject(ProjectsRepository projectsRepository);
}

File: app/src/main/java/com/example/mvvmgithubsampleapp/di/AppComponent.kt
package com.example.mvvmgithubsampleapp.di

import com.example.mvvmgithubsampleapp.repository.ProjectsRepository
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [AppModule::class])
interface AppComponent {
    fun inject(projectsRepository: ProjectsRepository)
}